tests/ServerSpec.hs

```diff
@@ -162,26 +179,33 @@ recipesChangesTest2 = do
     limits_ok :: RecipesChangesResponse -> Bool
     limits_ok response = rcrOffset response == 0 && rcrLimit response == 20
 
--- Check that limit and offset are parsed
--- XXX After /recipes/new has been added this will be used to test it more extensively
+-- | Check that limit and offset are parsed correctly
+--
+-- This must be called after posting > 10 changes to the recipe
 recipesChangesTest3 :: ClientM Bool
 recipesChangesTest3 = do
-    response <- getRecipesChanges "http-server" (Just 15) (Just 5)
-    return $ limits_ok response
+    response <- getRecipesChanges "A Test Recipe" (Just 5) (Just 5)
+    return $ limits_ok response && name_ok response && total_ok response
   where
     limits_ok :: RecipesChangesResponse -> Bool
-    limits_ok response = rcrOffset response == 15 && rcrLimit response == 5
+    limits_ok response = rcrOffset response == 5 && rcrLimit response == 5
 
+    name_ok :: RecipesChangesResponse -> Bool
+    name_ok response = rcName (rcrRecipes response !! 0) == "A Test Recipe"
 
--- | Start the app in a temporary directory for the recipes
+    total_ok :: RecipesChangesResponse -> Bool
+    total_ok response = rcTotal (rcrRecipes response !! 0) == 5
+
```

Comment:
Is it possible for `rcrRecipes` to be empty in these two lines?  I suppose it doesn't matter as much because this is a test, but still.

Reply:
Yes, it could be empty. I'll add a length test before this test.